runtime/vm/BytecodeInterpreter.hpp

```diff
@@ -3981,24 +3981,75 @@ class INTERPRETER_CLASS
 	}
 
 #if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
-	/* ToDo: unimplemented - https://github.com/eclipse-openj9/openj9/issues/13614 */
 	/* jdk.internal.misc.Unsafe: public native <V> V getValue(Object obj, long offset, Class<?> clz); */
 	VMINLINE VM_BytecodeAction
 	inlUnsafeGetValue(REGISTER_ARGS_LIST)
 	{
-		updateVMStruct(REGISTER_ARGS);
-		Assert_VM_unreachable();
-		return EXECUTE_BYTECODE;
+		j9object_t clz = *(j9object_t*)_sp;
+		I_64 offset = *(I_64*)(_sp + 1);
+		j9object_t obj = *(j9object_t*)(_sp + 3);
+
+		j9object_t result = NULL;
+		VM_BytecodeAction rc = EXECUTE_BYTECODE;
+
+		/* TODO (#14073): update this function to have the same behavior as OpenJDK when obj is null, clz is null, or when clz is not a VT class (currently OpenJDK segfaults in all of these scenarios) */
+		if (NULL != obj && NULL != clz) {
```

Comment:
You are missing the fast path allocation here.

Reply:
I think a simple way to handle this is to take parts out of  the `ValueTypeHelpers::getFlattenableField` function.

this part
```
			if (fastPath) {
				returnObjectRef = objectAllocate.inlineAllocateObject(currentThread, flattenedFieldClass, false, false);
				if (NULL == returnObjectRef) {
					goto done;
				}
			} else {
				VM_VMHelpers::pushObjectInSpecialFrame(currentThread, receiver);
				returnObjectRef = vm->memoryManagerFunctions->J9AllocateObject(currentThread, flattenedFieldClass, J9_GC_ALLOCATE_OBJECT_NON_INSTRUMENTABLE);
				receiver = VM_VMHelpers::popObjectInSpecialFrame(currentThread);
				if (J9_UNEXPECTED(NULL == returnObjectRef)) {
					goto done;
				}
				flattenedFieldClass = VM_VMHelpers::currentClass(flattenedFieldClass);
			}

			if (J9CLASS_HAS_4BYTE_PREPADDING(flattenedFieldClass)) {
				returnObjectOffset += sizeof(U_32);
			}

			objectAccessBarrier.copyObjectFields(currentThread,
								flattenedFieldClass,
								receiver,
								cache->offset + objectHeaderSize,
								returnObjectRef,
								objectHeaderSize + returnObjectOffset);
```

and make it a separate helper function and just call that here. That way you dont need to duplicate code.

Reply:
Okay, I'll get working on this now

Reply:
addressed in bcc3d7b